benchmarks/semi_coherent/Snakefile

```diff
@@ -0,0 +1,46 @@
+import os
+
+rule semi_coherent_sim:
+	input:
+		hepmcfile="benchmarks/semi_coherent/input.hepmc",
```

Comment:
Could you provide the script that does the filtering starting from some input that is available on S3?

We had used some hardcoded input files before, in https://eicweb.phy.anl.gov/EIC/datasets , but they never get updated and are a bit of a black box.

Reply:
Hi,

    In the following attachments I included the python scipt that I used for filtering the semi-coherent events from the original BeAGLE file after hepmc transformation, and a modified Snakefile that uses S3 data and applying the filtering script with afterburner.
    
[semi_coherent.zip](https://github.com/user-attachments/files/16515772/semi_coherent.zip)

    However I did encounter a problem trying to apply afterburner to the filtered `ePb_beagle.hepmc` file that was in `eictest/EPIC/EVGEN/CI`, and we think that the file might be too old. We would like to know if there is any other newer BeAGLE files on S3 with afterburner applied. If not can we upload our own files to S3.

Reply:
I've looked at the semi_coherent/Snakefile.py from the zip file. You are manually modifying text format for HepMC (v3?). That will not work with files on S3, as they are in the ROOT HepMC format. Could you rewrite that script using the HepMC bindings?

Here is an example snippet on how to read a ROOT HepMC event in python:
https://github.com/eic/detector_benchmarks/blob/a40a48bbfc0e0db6932cd22d96b5219cfc7879c2/benchmarks/backgrounds/ecal_backwards.org?plain=1#L92-L97
Writing should be similar (you could write to text format, if you want).

Reply:
Hi Dmitry,

I have just updated this branch to grab and filter the .hepmc file directly from S3. The beagle file in S3 is in hepmc format and not in ROOT hepmc format.